### ISSUE TITLE
[Feature] Improve run time of package with additional ftiler

### ISSUE THREAD

**Issue by CraigWilson-ZOE** [fivetran/dbt_mixpanel#38]
### Is there an existing feature request for this?

- [X] I have searched the existing issues

### Describe the Feature

When running within BigQuery I noticed that the model `stg_mixpanel__event_tmp` did not apply the same filter that the `mixpanel__event` model does i.e.:

```sql
where time >= {{ "'" ~ var('date_range_start',  '2010-01-01') ~ "'" }}
```

Applying this filter in the view can significantly reduce the resource requirements for running the model. When running in BigQuery for our own data, a run of the original code would process 4.45TB of data. Applying this filter, with the configuration below in our dbt_project.yml file, changed the processing to 276GB, which is a drastic improvement.

This is especially relevant when running in development, where you can use jinja to limit the number of records when running in development, which can greatly improve the time to test etc. e.g.

```yaml
vars:
   mixpanel:
      date_range_start: "{{ '2019-11-01' if target.name == 'production' else (modules.datetime.date.today() - modules.datetime.timedelta(7)) }}"
```

I believe applying the same variable based filtering in the temporary table, when this is used as the source of the `mixpanel__event` model is required to ensure the same amount of data is processed throughout the model.

### Are you interested in contributing this feature?

- [X] Yes.
- [ ] Yes, but I will need assistance and will schedule time during your [office hours](https://calendly.com/fivetran-solutions-team/fivetran-solutions-team-office-hours) for guidance.
- [ ] No.

**Comment by fivetran-avinash** [comment]
Hi @CraigWilson-ZOE ! Thanks for bringing this to our attention, I think you made a great point as to what the approach should be here. We agree that reducing package run time should be crucial in your development sandbox, particularly if you are not planning to use this data. 

Also, [thanks for submitting the proposed PR](https://github.com/fivetran/dbt_mixpanel/issues/38)! We will review this PR in a future sprint and see if there are more holistic changes we can make to the model to meet resourcing requirements in running dbt.

We will keep you in the loop about further updates here, and changes we plan to make to optimize package runs.

**Comment by fivetran-catfritz** [comment]
Hello @CraigWilson-ZOE, thank you for your suggestion regarding the filter. I am working on enhancing the overall performance of this Mixpanel package and have added additional updates on a test branch. I've incorporated your suggested upstream filter. In addition, I've eliminated the `stg_mixpanel__event_tmp` model in favor of transforming `stg_mixpanel__event` into an incremental model. I've also made adjustments to some of the clustering aspects and updated the incremental strategies for all the models. 

If you would be able to try out the test branch and let me know your thoughts and feedback, it would be appreciated! I am curious to know how much of an impact these updates make. Thank you!

To install the test branch, you can use the following in your packages.yml:
```yml
packages:
  - git: https://github.com/fivetran/dbt_mixpanel.git
    revision: feature/performance-enhancement
    warn-unpinned: false
```

**Comment by fivetran-catfritz** [comment]
Hi @CraigWilson-ZOE, following up to my last message, we still removed the tmp view, but decided to keep the stg model as ephemeral. This prevents duplicate tables while also allowing the start date to be set by the user. I have gone ahead and released [v0.9.0](https://github.com/fivetran/dbt_mixpanel/releases/tag/v0.9.0) and will mark this issue as closed, however if you have any additional feedback, please feel free to tag us in this thread or open a new issue!
